src/Workspaces/Core/Desktop/Workspace/Storage/PersistentStorageService.cs

```diff
@@ -284,10 +255,7 @@ public void RegisterPrimarySolution(SolutionId solutionId)
             // created when first C#/VB project is added
             lock (_lookupAccessLock)
             {
-                Contract.ThrowIfTrue(_primarySolutionStorage != null);
```

Comment:
this is existing code that killed host. it didn't check whether solution is same as _primarySolutionId